### PR TITLE
Rai pipeline sample

### PR DESCRIPTION
**Pull request by eniac871**
# PR into Azure/azureml-examples

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: sdk/jobs/pipelines/3a_rai_pipeline_sample/rai_components/component_causal.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    default: 1
+  random_state:
+    type: string # int or none
+    default: None
+
+outputs:
+  causal:
+    type: path
+
+code: ./rai_analyse/
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/3a_rai_pipeline_sample/rai_components/component_causal.yaml:58]: can we reuse the same code as CLI?

File: cli/jobs/pipelines-with-components/rai_pipeline_adult_analyse/components/component_gather_rai_insights.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+$schema: http://azureml/sdk-2-0/CommandComponent.json
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/rai_pipeline_adult_analyse/components/component_gather_rai_insights.yaml:1]: let's update these schema to clickable urls

File: .github/workflows/cli-jobs-pipelines-with-components-rai_pipeline_adult_analyse.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: cli-jobs-pipelines-with-components-rai_pipeline_adult_analyse
+on:
```
Comment (wangchao1230) [inline .github/workflows/cli-jobs-pipelines-with-components-rai_pipeline_adult_analyse.yml:2]: need to regenerate read me to adopt these new samples

File: sdk/jobs/pipelines/3a_rai_pipeline_sample/3a_rai_pipeline_sample.ipynb [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+   "metadata": {},
+   "source": [
+    "# Next Steps\n",
+    "You can see further examples of running a pipeline job [here](/sdk/jobs/pipelines/)"
+   ]
+  }
+ ],
+ "metadata": {
+  "description": {
+   "description": "Create pipeline with dsl.command_component"
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/3a_rai_pipeline_sample/3a_rai_pipeline_sample.ipynb:428]: need update metadata: description

File: sdk/jobs/pipelines/3a_rai_pipeline_sample/3a_rai_pipeline_sample.ipynb [hunk 5]
```diff
@@ -0,0 +1,454 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Build pipeline with dsl.command_component\n",
+    "\n",
+    "**Requirements** - In order to benefit from this tutorial, you will need:\n",
+    "- A basic understanding of Machine Learning\n",
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/3a_rai_pipeline_sample/3a_rai_pipeline_sample.ipynb:10]: need update the description

File: sdk/jobs/pipelines/3a_rai_pipeline_sample/3a_rai_pipeline_sample.ipynb [hunk 6]
```diff
@@ -0,0 +1,454 @@
+{
+ "cells": [
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/3a_rai_pipeline_sample/3a_rai_pipeline_sample.ipynb:2]: Need a readme file in this folder

File: cli/jobs/pipelines-with-components/rai_pipeline_adult_analyse/components/component_gather_rai_insights.yaml (hunk 2 above)
Comment (eniac871) [reply cli/jobs/pipelines-with-components/rai_pipeline_adult_analyse/components/component_gather_rai_insights.yaml]: updated

**Pull request merged**